### ISSUE TITLE
Question about the multi-gpu testing

### ISSUE THREAD

**Issue by PeiqinZhuang** [vt-vl-lab/video-data-aug#2]
Hi,

I have noticed that you recommend not using multi-GPU testing when using multi-GPU training.  Does the multi-GPU training mean the process of training?

**Comment by Yuliang-Zou** [comment]
Yes. You will need to change the port you use if you want to use multi-GPU testing during multi-GPU training.

**Comment by PeiqinZhuang** [comment]
> Yes. You will need to change the port you use if you want to use multi-GPU testing during multi-GPU training.

However, when using multi-GPU training, the evaluation hook for testing is DistEvalHook (shown in train_model.py), which means the evaluation is multi-GPU testing by default. The training process may be relatively slow when using multi-GPU testing. Is it right? To summarize, I think the default setting for evaluation during the training process is multi-GPU testing in this repo. How can I avoid this time-consuming strategy?

**Comment by Yuliang-Zou** [comment]
No. That is validation (launched in the same script with training). The multi-GPU testing I am referring to is launching another job in the same machine when you are training something. You can always disable the validation during training by removing the `--validate` in the training command.

**Comment by PeiqinZhuang** [comment]
> No. That is validation (launched in the same script with training). The multi-GPU testing I am referring to is launching another job in the same machine when you are training something. You can always disable the validation during training by removing the `--validate` in the training command.

Hi, I have figured out the bottleneck of training efficiency. Since the scale of the labeled dataset is small, the iterator for the labeled dataset would be generated several times. And this factor brings remarkable time consumptions in destroying the workers allocated before. There is a new feature in Pytorch1.9, in which a flag named persistent_workers is proposed to remain workers after each epoch.  With this new feature, the training process will be faster than before.

**Comment by Yuliang-Zou** [comment]
Cool. Do you want to make a PR?

**Comment by PeiqinZhuang** [comment]
OK. And would you mind uploading the config for Table 2. Right now I just replace the backbone with ResNet18(3D) with the config of ResNet34(2+1D). However, the results are relatively low. Since most people would be interested in Table 2, it would be appreciated if those configs will be uploaded.
